### PR TITLE
Refactor Fed configuration system.

### LINKED ISSUE #85: Refactor fed configuration system.
Currently, the Fed Configuration system is mess. It's not easy to introduce a new configuration item. To introduce a new item, we should change many places, such as putting a new item(k-v pair) to `internal kv`, fetching via internal_kv, and passing to other places.

For further developing, we'd like to refactor the fed configuration system, to let developers have a simple operation for that.

In the new system, we'd like to introduce `cluster configurations` and `job configurations`. All of the above will be written into internal kv.

### PR DESCRIPTION
**Pull request by jovany-wang**
Close #85

### CODE REVIEW

File: fed/api.py [hunk 1]
```diff
@@ -203,38 +213,33 @@ def shutdown():
...
     ray.shutdown()
     logger.info('Shutdowned ray.')
 
 
-def get_cluster():
+def _get_cluster():
     """
     Get the RayFed cluster configration.
     """
     # TODO(qwang): These getter could be cached in local.
```
Comment (NKcqx) [inline fed/api.py:227]: I think this is done.

Comment (jovany-wang) [reply fed/api.py:227]: Gooood catch!

File: fed/api.py [hunk 2]
```diff
@@ -160,13 +163,20 @@ def init(
 
     compatible_utils._init_internal_kv()
     compatible_utils.kv.initialize()
```
[the remark this answers is not in the document]
Comment (jovany-wang) [reply fed/api.py:165]: I don't think so. We didn't initialize it in this code path.

[the remark this answers is not in the document]
Comment (NKcqx) [reply fed/api.py:165]: u r right. Here initialize the kv, and the below's initialization only happens when putting `ClusterConfig`.

**Review by NKcqx — APPROVED** [review]
LGTM

**Pull request merged**